libraries/shared/watcher/storage_watcher.go

```diff
@@ -73,21 +79,31 @@ func (storageWatcher StorageWatcher) Execute(diffsChan chan utils.StorageDiff, e
 	}
 }
 
-func (storageWatcher StorageWatcher) getTransformer(diff utils.StorageDiff) (transformer.StorageTransformer, bool) {
+func (storageWatcher StorageWatcher) getTransformer(diff utils.PersistedStorageDiff) (transformer.StorageTransformer, bool) {
 	storageTransformer, ok := storageWatcher.KeccakAddressTransformers[diff.HashedAddress]
 	return storageTransformer, ok
 }
 
-func (storageWatcher StorageWatcher) processRow(diff utils.StorageDiff) {
-	storageTransformer, ok := storageWatcher.getTransformer(diff)
+func (storageWatcher StorageWatcher) processRow(diffInput utils.StorageDiffInput) {
+	diffID, err := storageWatcher.StorageDiffRepository.CreateStorageDiff(diffInput)
+	if err != nil {
+		if err == repositories.ErrDuplicateDiff {
+			logrus.Warn("ignoring duplicate diff")
+			return
+		}
+		logrus.Warnf("failed to persist storage diff: %s", err.Error())
+		// TODO: bail? Should we continue attempting to transform a diff we didn't persist
```

Comment:
errrrr, tricky. That would invalidate the expected data model in the DB right? Can we take the time to at least retry?

Reply:
Like, what are the potential reasons for this failure?